tests/core/nlg/test_response.py

```diff
@@ -0,0 +1,295 @@
+    tracker = DialogueStateTracker(sender_id="conversation_id", slots=[slot])
+
+    r = await t.generate(
+        utter_action="utter_withdraw", tracker=tracker, output_channel=""
+    )
+    assert r.get("text") == bot_message
+
+
+async def test_nlg_conditional_response_variations_with_yaml_multi_constraints():
+    domain = Domain.from_file(
+        path="data/test_domains/conditional_response_variations.yml"
+    )
+    t = TemplatedNaturalLanguageGenerator(responses=domain.responses)
+
+    first_slot = Slot(
+        name="account_type", initial_value="primary", influence_conversation=False
+    )
+    second_slot = Slot(
+        name="can_withdraw", initial_value=True, influence_conversation=False
+    )
+    tracker = DialogueStateTracker(
+        sender_id="conversation_id", slots=[first_slot, second_slot]
+    )
+    r = await t.generate(
+        utter_action="utter_withdraw", tracker=tracker, output_channel=""
+    )
+    assert r.get("text") == "Withdrawal has been approved."
+
+
+async def test_nlg_conditional_response_variations_with_yaml_and_channel():
+    domain = Domain.from_file(
+        path="data/test_domains/conditional_response_variations.yml"
+    )
+    t = TemplatedNaturalLanguageGenerator(responses=domain.responses)
+
+    slot = Slot(
+        name="account_type", initial_value="primary", influence_conversation=False
+    )
+    tracker = DialogueStateTracker(sender_id="conversation_id", slots=[slot])
+
+    r = await t.generate(
+        utter_action="utter_check_balance", tracker=tracker, output_channel="os"
+    )
+    assert (
+        r.get("text")
+        == "As a primary account holder, you can now set-up your access on mobile app too."
+    )
+
+    resp = await t.generate(
+        utter_action="utter_check_balance", tracker=tracker, output_channel="app"
+    )
+    assert resp.get("text") == "Welcome to your app account overview."
+
+
+@pytest.mark.parametrize(
+    ("slot_name", "slot_value", "message"),
+    (
+        ("test_bool", True, "example boolean"),
+        ("test_int", 12, "example integer"),
+        ("test_list", [], "example list"),
+    ),
+)
+async def test_nlg_conditional_response_variations_with_diff_slot_types(
+    slot_name: Text, slot_value: Any, message: Text
+):
+    responses = {
+        "utter_action": [
+            {
+                "text": "example boolean",
+                "condition": [{"type": "slot", "name": "test_bool", "value": True}],
+            },
+            {
+                "text": "example integer",
+                "condition": [{"type": "slot", "name": "test_int", "value": 12}],
+            },
+            {
+                "text": "example list",
+                "condition": [{"type": "slot", "name": "test_list", "value": []}],
+            },
+        ]
+    }
+    t = TemplatedNaturalLanguageGenerator(responses=responses)
+    slot = Slot(name=slot_name, initial_value=slot_value, influence_conversation=False)
```

Comment:
this uncovers a faulty piece in our code. `Slot` is actually an abstract class and shouldn't be instantiated (the methods raising `NotImplementedError` should be marked with `abc.abstractmethod`). Out of scope of your PR but if you truly want to test different slot types you should use things like `TextSlot` or `ListSlot`